### DISCUSSION TITLE
std::ops::Neg for matrix types?

### DISCUSSION THREAD

**Discussion by chubei-oppen** [bitshifter/glam-rs#256] [Ideas]
The title says it all.

EDIT: Looks like as easy as a trivial impl. I tried one in https://github.com/chubei-oppen/glam-rs/commit/0a3804eee3c2f6719ec56919b3c275ff6d6b5043.

**Comment by bitshifter** [comment]
Yeah that would be easy enough to add. I'm curious what you need it for?

If you want to make a PR go for it, would need to add it for `Mat2` and `Mat4` as well and ideally add tests.

**Reply by chubei-oppen** [reply]
To calcuate the inverse of a rigid body transform:

```
[R t]
[0 1]
```

I want to write:

```
[R.transpose -R.transpose * t]
[0           1               ]
```

Of course I can write `R.transpose * (-t)` but that's an extra pair of parentheses.

Can you kindly point out where the tests should be added? There are so many macros.

**Comment by bitshifter** [comment]
They could go in
https://github.com/bitshifter/glam-rs/blob/main/tests/mat3.rs#L287. There
are equivalent tests for mat2 and mat4.

**Comment by chubei-oppen** [comment]
Implemented in #258
